search.html

```diff
@@ -223,49 +231,163 @@
 // search logic
 var execute_search = function() {
     // get search options
-    var expansion = $("input[name=expansion]:checked").val() == "expansion" ? true : false;
-    var boolean = $("input[name=boolean]:checked").val();
-    var search_type = $("select[name=type-facet]").val();
-    var fields = {};
-
-    $("input[name=scope]:checked").each(function () {
-        fields[this.value] = {};
-    });
-    search_options["expand"] = expansion;
-    search_options["fields"] = fields;
-    // search_options["boolean"] = boolean;
-    search_tags = $("#tag-facet").select2("data").map(function(datum){
-      return datum.text;
-    });
+
+    var search_options = { "fields": { "content": { "expand": true } } };
+    var type_counts = {};
+    var tag_counts = {};
 
     // get search query
     var query = $('#search-input').val();
+    debugger
+
+    // if (search_query === undefined || $('#search-input').val() !== "") {
+    //   query = $('#search-input').val();
+    // } else {
+    //   query = search_query;
+    //   $('#search-input').val(search_query);
+    // }
 
     // empty result container when search query is empty
     if (query === '') {
-        jQuery('.search-results').empty();
+      $('#search-tip').css('display', 'none');
+      $('#type-filter').css('display', 'none');
+      $('#tag-filter').css('display', 'none');
+      $('#search-results').empty();
     } else {
         // build the index
-        var index = build_index(elasticlunr_index, search_type, search_tags);
+        var index = build_index(elasticlunr_index, search_type, search_tag);
 
         // perform search
         var results = index.search(query, search_options);
 
+        // collect type counts
+        results.map(function(result) {
+          if (type_counts[store[result.ref].type] == undefined) {
+            type_counts[store[result.ref].type] = 1;
+          } else {
+            type_counts[store[result.ref].type] = type_counts[store[result.ref].type] + 1;
+          }
+        });
+
+        // collect tag counts
+        results.map(function(result) {
+          if (store[result.ref].tags !== undefined && store[result.ref].tags.length != 0) {
+            for (var i = 0; i < store[result.ref].tags.length; i++) {
+              if (tag_counts[store[result.ref].tags[i]] === undefined) {
+                tag_counts[store[result.ref].tags[i]] = 1;
+              } else {
+                tag_counts[store[result.ref].tags[i]] = tag_counts[store[result.ref].tags[i]] + 1;
+              }
+            }
+          }
+        });
+
+        // apply search filters and display in the UI
```

Comment:
This seems a bit complicated, do we need all of it?  Or can we at least consolidate with the repeated parts in the second block?